modal/_blob_utils.py

```diff
@@ -322,13 +322,14 @@ class FileUploadSpec:
     use_blob: bool
     content: Optional[bytes]  # typically None if using blob, required otherwise
     sha256_hex: str
+    mode: int  # file permission bits (last 12 bits of st_mode)
     size: int
 
 
 def get_file_upload_spec(filename: Path, mount_filename: str) -> FileUploadSpec:
     # Somewhat CPU intensive, so we run it in a thread/process
-    size = os.path.getsize(filename)
-    if size >= LARGE_FILE_LIMIT:
+    stat = os.stat(filename)
```

Comment:
Does this work on Windows?

Reply:
Yeah, it does appear to, as of at least Python 3.3.4. https://stackoverflow.com/questions/5275731/os-stat-on-windows

I can add a client test to make sure also

Reply:
Thanks for pointing it out! I ended up doing this, which should be fully backwards-compatible.

```python
        # Python appears to give files 0o666 bits on Windows (equal for user, group, and global),
        # so we mask those out to 0o755 for compatibility with POSIX-based permissions.
        mode=stat.st_mode & (0o7777 if platform.system() != "Windows" else 0o7755),
```

By the way, the mode flag needs to be included in the mount content hash. But we need to do this in a backwards-compatible way on the client mount to not trigger a bunch of image rebuilds, so I'm just skipping that field in the current JSON-based hash format if the mode is equal to `0o644`.

I just checked our client mount, and there are no files with executable bit set, luckily.